tests/components/enphase_envoy/test_sensor.py

```diff
@@ -37,3 +42,804 @@ async def test_sensor(
     with patch("homeassistant.components.enphase_envoy.PLATFORMS", [Platform.SENSOR]):
         await setup_integration(hass, config_entry)
     await snapshot_platform(hass, entity_registry, snapshot, config_entry.entry_id)
+
+
+PRODUCTION_NAMES: tuple[str, ...] = (
+    "current_power_production",
+    "energy_production_today",
+    "energy_production_last_seven_days",
+    "lifetime_energy_production",
+)
+
+
+@pytest.mark.parametrize(
+    ("mock_envoy"),
+    [
+        "envoy",
+        "envoy_1p_metered",
+        "envoy_metered_batt_relay",
+        "envoy_nobatt_metered_3p",
+        "envoy_tot_cons_metered",
+    ],
+    indirect=["mock_envoy"],
+)
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_sensor_production_data(
+    hass: HomeAssistant,
+    mock_envoy: AsyncMock,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test production entities values."""
+    with patch("homeassistant.components.enphase_envoy.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, config_entry)
+
+    sn = mock_envoy.serial_number
+    ENTITY_BASE: str = f"{Platform.SENSOR}.envoy_{sn}"
+
+    data = mock_envoy.data.system_production
+    PRODUCTION_TARGETS: tuple[float, ...] = (
+        data.watts_now / 1000.0,
+        data.watt_hours_today / 1000.0,
+        data.watt_hours_last_7_days / 1000.0,
+        data.watt_hours_lifetime / 1000000.0,
+    )
+
+    for name, target in list(zip(PRODUCTION_NAMES, PRODUCTION_TARGETS, strict=False)):
+        assert (entity_state := hass.states.get(f"{ENTITY_BASE}_{name}"))
+        assert target == float(entity_state.state)
```

Comment:
Please put the expected value in the assertion on the right hand side.

Reply:
Thanks for the review, I've implemented it in PR #122726.